src/Uno.Foundation/IPropertyValue.cs

```diff
@@ -1,133 +1,245 @@
-		/// <param name="value">The array of values.</param>
-		void GetSingleArray(out float[] value);
-
-		void GetDoubleArray(out double[] value);
-
-		void GetChar16Array(out char[] value);
-
-		void GetBooleanArray(out bool[] value);
-
-		/// <summary>Returns the array of string values stored as a property value.</summary>
-		/// <param name="value">The array of values.</param>
-		void GetStringArray(out string[] value);
-
-		/// <summary>Returns the array of inspectable objects stored as a property value.</summary>
-		/// <param name="value">The array of objects.</param>
-		void GetInspectableArray(out object[] value);
-
-		void GetGuidArray(out Guid[] value);
-
-		void GetDateTimeArray(out DateTime[] value);
-
-		/// <summary>Returns the array of time interval values stored as a property value.</summary>
-		/// <param name="value">The array of values.</param>
-		void GetTimeSpanArray(out TimeSpan[] value);
-
-		/// <summary>Returns the array of point structures stored as a property value.</summary>
-		/// <param name="value">The array of structures.</param>
-		void GetPointArray(out Point[] value);
-
-		/// <summary>Returns the array of size structures stored as a property value.</summary>
-		/// <param name="value">The array of structures.</param>
-		void GetSizeArray(out Size[] value);
-
-		/// <summary>Returns the array of rectangle structures stored as a property value.</summary>
-		/// <param name="value">The array of structures.</param>
-		void GetRectArray(out Rect[] value);
-	}
+	/// <summary>
+	/// Gets a value that indicates whether the property value is a scalar value.
+	/// </summary>
+	/// <returns>
+	/// True if the value is scalar; otherwise false.
+	/// </returns>
+	bool IsNumericScalar { get; }
+
+	/// <summary>
+	/// Returns the type stored in the property value.
+	/// </summary>
+	PropertyType Type { get; }
+
+	/// <summary>
+	/// Returns the Boolean value stored as a property value.
+	/// </summary>
+	/// <returns>The value.</returns>
+	bool GetBoolean();
+
+	/// <summary>
+	/// Returns the array of Boolean values stored as a property value.
+	/// </summary>
+	/// <param name="value">The array of values.</param>
+	void GetBooleanArray(out bool[] value);
+
+	/// <summary>
+	/// Returns the Unicode character stored as a property value.
+	/// </summary>
+	/// <returns>The value.</returns>
+	char GetChar16();
+
+	/// <summary>
+	/// Returns the array of Unicode characters stored as a property value.
+	/// </summary>
+	/// <param name="value">The array of values.</param>
+	void GetChar16Array(out char[] value);
+
+	/// <summary>
+	/// Returns the date and time value stored as a property value.
+	/// </summary>
+	/// <returns>The value.</returns>
+	DateTimeOffset GetDateTime();
+
+	/// <summary>
+	/// Returns the array of date and time values stored as a property value.
+	/// </summary>
+	/// <param name="value">The array of values.</param>
+	void GetDateTimeArray(out DateTime[] value);
```

Comment:
Should the parameter here be `DateTimeOffset[]` instead of `DateTime[]`?

Reply:
Indeed, there was a bug in UWP/WinUI docs for this! Fixed the code